gips/data/landsat/landsat.py

```diff
@@ -286,6 +289,53 @@ def __init__(self, filename):
             raise Exception("Sensor %s not supported: %s" % (self.sensor, filename))
         self._version = self.version
 
+    def filter(self, pclouds=100, **kwargs):
+        if pclouds < 100:
+            if os.path.exists(self.filename):
+                mtlfilename = [f for f in self.datafiles() if 'MTL.txt' in f][0]
+                mtlfilename = self.extract([mtlfilename])[0]
```

Comment:
Optional:  Suggest combining the lines to cut down on `[0]` and avoid name re-purposing (I'm generally opposed to re-using names):

```
mtlfilename = self.extract([f for f in self.datafiles() if 'MTL.txt' in f])[0]
```

if you want you can also `f.endswith('MTL.txt')` for extra precision.